### PR TITLE
Fix: Include quotes when identifier has space, dot, or dash

### LINKED ISSUE #296: $__timeFilter on a column that has dashes in its name does not work
**What happened**:

I get the following weird error when using query builder in grafana with azure data explorer data source:

```
Failed to execute query: Azure HTTP "400 BadRequest": 
"Request is invalid and cannot be processed: Syntax error: SYN0002: A recognition error occurred. [line:position=2:107]. 
Query: '['fridge-telemetry-from-hub']\n| where TimeGenerated >= datetime(2021-11-20T23:00:00Z) and TimeGenerated <= datetime(2021-11-22T22:59:59Z)(messageProperties_iothub-creation-time-utc)\n| summarize by '"
```

**What you expected to happen**:

The query builder would work and return results from the underlying adx resource.

**How to reproduce it (as minimally and precisely as possible)**:

See the screenshot below. Likely have an adx table with datetime column that has dashes in it's name.

**Screenshots**

image

**Anything else we need to know?**:

I think the issue is in this regexp, the column names that have dashes do not work with the ``$__timeFilter(column-name)`` syntax:

https://github.com/grafana/azure-data-explorer-datasource/blob/919c9dbdeae97589ceb3ef2d70a1d07fd7174c9e/pkg/azuredx/models/macro.go#L36-L38

image

**Environment**:

- Grafana version: Grafana v8.3.3 (30bb7a93ca)
- Plugin version: 3.6.1
- OS Grafana is installed on: linux container
- User OS & Browser: Windows 10, Edge
- Others:

### PR DESCRIPTION
**Pull request by fridgepoet**
Need to at least quote identifiers with spaces, dots, or dashes.
https://docs.microsoft.com/en-us/azure/data-explorer/kusto/query/schema-entities/entity-names#identifier-quoting

Fixes #296

### CODE REVIEW

File: pkg/azuredx/models/macro.go [hunk 1]
```diff
@@ -35,7 +35,7 @@ func NewMacroData(tr *backend.TimeRange, intervalMS int64) MacroData {
 // macroRE is a regular expression to match available macros
 var macroRE = regexp.MustCompile(`\$__` + // Prefix: $__
 	`(timeFilter|timeFrom|timeTo|timeInterval)` + // one of macro root names
-	`(\([a-zA-Z0-9_-]*?\))?`) // optional () or optional (someArg)
+	`(\([a-zA-Z0-9_ .-]*?\))?`) // optional () or optional (someArg)
```
Comment (andresmgot) [inline pkg/azuredx/models/macro.go:38]: nit: for readability I would use `\s\.`

File: pkg/azuredx/models/macro.go [hunk 2]
```diff
@@ -66,6 +66,13 @@ func (md MacroData) Interpolate(query string) (string, error) {
 	return interpolated, nil
 }
 
+func quoteForSpacesDotsDashes(s string) string {
+	if strings.ContainsAny(s, " .-") {
```
Comment (andresmgot) [inline pkg/azuredx/models/macro.go:70]: can you add a link to the docs you found here as a comment? (for future readers)

**Review by andresmgot — APPROVED** [review]
Great, thanks!

File: pkg/azuredx/models/macro.go (hunk 1 above)
Comment (fridgepoet) [reply pkg/azuredx/models/macro.go]: ok! how about with just `\s.-`?

**Review by andresmgot — APPROVED** [review]

**Pull request merged**